src/components/Map/PopupComponent.tsx

```diff
@@ -0,0 +1,36 @@
+import { type SensorData } from 'src/models/SensorData';
+
+interface PopupComponentProps {
+  sensorData: SensorData;
+}
+
+function PopupComponent({ sensorData }: PopupComponentProps) {
+  return (
+    <div className='text-center'>
+      <h3 className='text-xl font-bold'>{sensorData.name}</h3>
+      <h4 className='text-left text-l font-bold'>
+        {sensorData.date.toLocaleDateString()}{' '}
+        {sensorData.date.toLocaleTimeString()}
+      </h4>
+      <table className='table-auto w-full mt-3'>
+        <thead>
+          <tr>
+            <th className='px-4 py-2'>Type</th>
+            <th className='px-4 py-2'>Value</th>
+          </tr>
+        </thead>
+        <tbody>
+          {sensorData.data.map((item, index) => (
+            // eslint-disable-next-line react/no-array-index-key
+            <tr key={index}>
```

Comment:
I think we will get some ids in responses from the backend for each `AirImpurityData` object so this key should be changed to something else than `index` in the future. Generally, using array index as a key is a bad practice and has impact on the performance. You can add some `TODO` comment not to forget about this change and leave it as it is for now or change key to the `name` value of the `AirImpurityData` object (if names are unique - I think, they should be).